### PR TITLE
docs(v2): update config docs

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

Update docs to clarify certain stuff especially on configuration. A lot of people is confused on `baseUrl` and `url`. 

Made some example simpler

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

- N/A

### CODE REVIEW

File: website/docs/docusaurus.config.js.md [hunk 1]
```diff
@@ -18,7 +14,7 @@ title: docusaurus.config.js
 
 - Type: `string`
 
-Title is used in a number of places in your site including the title for the web page, headings, etc.
```
Comment (endiliey) [inline website/docs/docusaurus.config.js.md]: Less word is better.

File: website/docs/configuration.md [hunk 2]
```diff
@@ -73,27 +71,20 @@ module.exports = {
...
+
+Basic Example:
 
 ```jsx
 import React from 'react';
-import Head from '@docusaurus/Head';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 
-const Layout = props => {
+const Hello = () => {
```
Comment (endiliey) [inline website/docs/configuration.md]: Simplify example

File: website/docs/docusaurus.config.js.md [hunk 3]
```diff
@@ -31,7 +27,7 @@ module.exports = {
 
 - Type: `string`
 
-Tagline is used in a number of places in your site including the title for the web page, sub headings, etc.
+The tagline for your website.
```
Comment (yangshun) [inline website/docs/docusaurus.config.js.md]: Actually is this field strictly necessary? If we don't use it in the pages/index it's not used right?

Comment (endiliey) [reply website/docs/docusaurus.config.js.md]: i think not really required, i think we can make it theme-specific. but It is used in theme-fallback layout now (which we can remove). Its only used as default title in @theme/Layout.

Comment (endiliey) [reply website/docs/docusaurus.config.js.md]: but this PR is only about docs change. I'm gonna take a rest. do you want to remove it from required field of docusaurus.config and make it as themeConfig instead ?

I think we're still missing a doc page for @docusaurus-theme-classic Theme Configuration
It's still not clear what are the fields available. eg: navbar, footer, image

**Pull request merged**